s3stream/src/main/java/com/automq/stream/s3/DeltaWALUploadTask.java

```diff
@@ -55,10 +58,18 @@ public class DeltaWALUploadTask {
     private long uploadTimestamp;
     private long commitTimestamp;
     private volatile CommitStreamSetObjectRequest commitStreamSetObjectRequest;
+    private volatile boolean burst = false;
+    private final long id;
+    private final LongAdder rateLimitTime = new LongAdder();
+    private final LongAdder burstBytes = new LongAdder();
+
+    public S3Storage.DeltaWALUploadTaskContext ctx;
+    public LongSupplier burstTaskId;
 
     public DeltaWALUploadTask(Config config, Map<Long, List<StreamRecordBatch>> streamRecordsMap,
-        ObjectManager objectManager, ObjectStorage objectStorage,
-        ExecutorService executor, boolean forceSplit, double rate) {
+                              ObjectManager objectManager, ObjectStorage objectStorage,
+                              ExecutorService executor, boolean forceSplit, double rate,
+                              long id, LongSupplier burstTaskId) {
```

Comment:
Why we need the id and burstTaskId here? DeltaWALUploadTask already has the #burst method

Reply:
removed.